### ISSUE TITLE
Mapper error when I try to execute QuantumCircuit as an object of a user defined class

### ISSUE THREAD

**Issue by isolatedinformation** [Qiskit/qiskit-terra#1820]
### Informations

- **Qiskit version**: 0.7.0
- **Python version**: 3.6.7
- **Operating system**: Ubuntu 18.04

### What is the current behavior?
/home/varun/.local/lib/python3.6/site-packages/marshmallow/schema.py:364: ChangedInMarshmallow3Warning: strict=False is not recommended. In marshmallow 3.0, schemas will always be strict. See https://marshmallow.readthedocs.io/en/latest/upgrading.html#schemas-are-always-strict
  ChangedInMarshmallow3Warning
Please provide the cooeficeints for the intital state in the format a+bj
Coefficient of state zero1+0j
Coefficient of state one0+0j
Provide the square of normalisation denominator1
[1.+0.j 0.+0.j]
The best backend is ibmq_16_melbourne
Traceback (most recent call last):
  File "class_counts.py", line 183, in <module>
    main()
  File "class_counts.py", line 163, in main
    sim = obj.simulate(p,b)
  File "class_counts.py", line 111, in simulate
    dec_state = self.ibmq(b)
  File "class_counts.py", line 97, in ibmq
    job_exp = execute(self.qc, backend=backend, shots=shots, max_credits=max_credits)
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/tools/compiler.py", line 108, in execute
    skip_transpiler, seed_mapper, pass_manager, memory)
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/tools/compiler.py", line 61, in compile
    seed_mapper, pass_manager)
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/transpiler/_transpiler.py", line 67, in transpile
    'pass_manager': pass_manager})
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/tools/parallel.py", line 93, in parallel_map
    return [task(values[0], *task_args, **task_kwargs)]
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/transpiler/_transpiler.py", line 109, in _transpilation
    pass_manager=pass_manager)
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/transpiler/_transpiler.py", line 199, in transpile_dag
    dag = Optimize1qGates().run(dag)
  File "/home/varun/.local/lib/python3.6/site-packages/qiskit/transpiler/passes/optimize_1q_gates.py", line 53, in run
    raise MapperError("internal error")
qiskit.mapper._mappererror.MapperError: 'internal error'

### Steps to reproduce the problem
The QuantumRegister, ClassicalRegister and QuantumCircuit are intialised when a object is created for the user-defined class I have made. The code ran without errors on the local machine and on the hpc. But when I tried to run it on the ibm Qx_16 it showed this error.

### What is the expected behavior?
I expect to get values of get_counts() for 20 different values for a given parameter. For each value a new object is created and the entire circuit is simulated.

### Suggested solutions
The problem could be because I create a new object every iteration of the loop. I made sure to delete the object after each iteration though,

**Comment by maddy-tod** [comment]
Hi @isolatedinformation could you provide some more information about the circuit you were trying to execute when this error occurred?

**Comment by isolatedinformation** [comment]
My circuit consisits of 9 qubits and 9 cbits to record thier outcomes. Basically, I was trying to simulate a noisy channel and perform error correction to retireve the initial message. The noisiness of the channel was tuned by a parameter \gamma. For diiferent values of \gamma, I wanted the statistics of the measuremnt outcome and plotted them. For instance, this attached image was the statistics I got for 20 different values of gamma when run on the HPC.
![zero](https://user-images.githubusercontent.com/27089492/52951977-3731a200-33a9-11e9-9c80-dc164d88c2b0.png)

I created a class to make sure the entire circuit was re initilased for every iteration of \gamma since each iteration is independent of the other. The objects of the class helped achieve this objective.

If it's still not clear, I can email my code to you.
